distrobox-create

```diff
@@ -59,7 +61,15 @@ container_user_gid="$(id -rg)"
 container_user_home="${HOME:-"/"}"
 container_user_name="${USER}"
 container_user_uid="$(id -ru)"
-distrobox_sudo_program="sudo"
+# If we're running distrobox-create as root - as in, logged in the shell as
+# root user, and not via sudo -, we don't want to set distrobox_sudo_program
+# (it's meaningless for our usecase).
+if [ "${container_user_uid}" -ne 0 ]; then
+	# If the DBX_SUDO_PROGRAM/distrobox_sudo_program variable was set by the
+	# user, use its value instead of "sudo". But only if not runnig the script
+	# as root (UID 0).
+	distrobox_sudo_program=${DBX_SUDO_PROGRAM:-${distrobox_sudo_program:-"sudo"}}
+fi
```

Comment:
If we move this block after the "source" command, it will continue to work as before, in the sense that a non-root user can set the sudo program in the config files also